src/reconcile/rec_row.c

```diff
@@ -751,6 +755,11 @@ __wt_rec_row_leaf(
      */
     WT_ERR(__wt_scr_alloc(session, 0, &tmpkey));
 
+    /* About to do some visibility checks. Update our snapshot for eviction. */
+    if (F_ISSET(session, WT_SESSION_INTERNAL) && F_ISSET(session->txn, WT_TXN_RUNNING) &&
+      F_ISSET(r, WT_REC_EVICT))
+        __wt_txn_bump_snapshot(session);
```

Comment:
Comment, explaining the choice of this API for refreshing the snapshot will be useful.

Reply:
I'll add appropriate comment.